File: src/main/java/kitchenpos/products/tobe/domain/ProductPriceChangedEvent.java
package kitchenpos.products.tobe.domain;

import org.springframework.context.ApplicationEvent;

import java.util.UUID;

public class ProductPriceChangedEvent extends ApplicationEvent {

    private UUID productId;

    private ProductPrice productPrice;

    public ProductPriceChangedEvent(Object source, UUID productId, ProductPrice productPrice) {
        super(source);
        this.productId = productId;
        this.productPrice = productPrice;
    }

    public UUID getProductId() {
        return productId;
    }

    public ProductPrice getProductPrice() {
        return productPrice;
    }
}

File: src/main/java/kitchenpos/products/tobe/domain/ProductPriceChangedEvent.kt
package kitchenpos.products.tobe.domain

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

data class ProductPriceChangedEvent(
    val productId: UUID,
    val oldPrice: BigDecimal,
    val newPrice: BigDecimal,
    override val occurredOn: LocalDateTime = LocalDateTime.now(),
) : DomainEvent
